src/care/util.h

```diff
@@ -299,9 +299,86 @@ namespace care {
    }
 #endif
 
+// various GPU wrappers, only needed for GPU compiles
+#if defined (CARE_GPUCC)
+
+// wrapper for hip/cuda free
+CARE_HOST inline void gpuFree(void* buffer) {
```

Comment:
These might have to be defined in a source file if you don't have the cuda/hip headers included in config.h.in.

Reply:
Not an issue, it seems that as long as we are compiling with the hipcc wrapper, I don't run into issues. (It might be different if we compiled with a regular c++ compiler and just included the HIP libraries.

To test this, I modified the code here to be

CARE_HOST inline void gpuFree(void* buffer) {
#if defined(ENABLE_HIP)
#warning GPUFREEHIP
   gpuAssert(hipFree(buffer), "gpuFree", __LINE__);
#elif defined(ENABLE_CUDA)
   gpuAssert(cudaFree(buffer), "gpuFree", __LINE__);
#endif
}

And the compile warning went off as expected, and there was no problem with hipFree